renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -163,9 +348,61 @@ struct VulkanPixelHistoryCallback : public VulkanDrawcallCallback
     m_pDriver->GetRenderState().EndRenderPass(cmd);
     // Get pre-modification values
     size_t storeOffset = m_EventIndices.size() * sizeof(EventInfo);
-    CopyPixel(eid, cmd, storeOffset);
+    VkImage depthImage = VK_NULL_HANDLE;
+    VkFormat depthFormat = VK_FORMAT_UNDEFINED;
+    const DrawcallDescription *draw = m_pDriver->GetDrawcall(eid);
+    if(draw && draw->depthOut != ResourceId())
+    {
+      ResourceId resId = m_pDriver->GetResourceManager()->GetLiveID(draw->depthOut);
+      depthImage = m_pDriver->GetResourceManager()->GetCurrentHandle<VkImage>(resId);
+      const VulkanCreationInfo::Image &imginfo = m_pDriver->GetDebugManager()->GetImageInfo(resId);
+      depthFormat = imginfo.format;
+    }
+
+    CopyPixel(m_Image, m_Format, depthImage, depthFormat, cmd, storeOffset);
 
     // TODO: add all logic for draw calls.
+    VulkanRenderState &pipestate = m_pDriver->GetRenderState();
+    ResourceId prevState = pipestate.graphics.pipeline;
+    ResourceId prevRenderpass = pipestate.renderPass;
+    ResourceId prevFramebuffer = pipestate.framebuffer;
+    uint32_t prevSubpass = pipestate.subpass;
+
+    {
+      auto pipeIt = m_PipelineCache.find(pipestate.graphics.pipeline);
+      PipelineReplacements replacements;
+
+      if(pipeIt != m_PipelineCache.end())
+        replacements = pipeIt->second;
```

Comment:
Minor style nit - if either branch of an `if` has braces please add them to both (i.e. here the `if` case should have `{}`).

Reply:
Done